### ISSUE TITLE
XML-LibXML test fails with Intel

### ISSUE THREAD

**Issue by vanzod** [easybuilders/easybuild-easyconfigs#7432]
I am hitting a strange error when building `XML-LibXML-2.0132-intel-2017b-Perl-5.26.0.eb` during the test step:

```
== 2019-01-15 10:08:24,596 run.py:192 INFO running cmd: make test 
== 2019-01-15 10:08:37,332 build_log.py:162 ERROR EasyBuild crashed with an error (at easybuild/EB-develop/vsc-base/lib/vsc/utils/exceptions.py:124 in __init__): cmd "make test" exited with exit code 2 and output:
"/opt/easybuild/software/Compiler/GCCcore/6.4.0/Perl/5.26.0/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- LibXML.bs blib/arch/auto/XML/LibXML/LibXML.bs 644
PERL_DL_NONLAZY=1 "/opt/easybuild/software/Compiler/GCCcore/6.4.0/Perl/5.26.0/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
Can't load '/mnt/ramdisk/XMLLibXML/2.0132/intel-2017b-Perl-5.26.0/XML-LibXML-2.0132/blib/arch/auto/XML/LibXML/LibXML.so' for module XML::LibXML: /mnt/ramdisk/XMLLibXML/2.0132/intel-2017b-Perl-5.26.0/XML-LibXML-2.0132/blib/arch/auto/XML/LibXML/LibXML.so: undefined symbol: __intel_ssse3_strncpy at /opt/easybuild/software/Compiler/GCCcore/6.4.0/Perl/5.26.0/lib/perl5/5.26.0/x86_64-linux-thread-multi/DynaLoader.pm line 193.
 at /mnt/ramdisk/XMLLibXML/2.0132/intel-2017b-Perl-5.26.0/XML-LibXML-2.0132/blib/lib/XML/LibXML.pm line 156.
BEGIN failed--compilation aborted at /mnt/ramdisk/XMLLibXML/2.0132/intel-2017b-Perl-5.26.0/XML-LibXML-2.0132/blib/lib/XML/LibXML.pm line 156.
Compilation failed in require at t/01basic.t line 6.
BEGIN failed--compilation aborted at t/01basic.t line 6.
```

Has anybody seen this issue before?

**Comment by boegel** [comment]
The missing symbol `__intel_ssse3_strncpy` is a part of one of the libraries provided by the Intel compilers.
See https://groups.google.com/forum/#!topic/enzo-users/QF17H6Sdans, looks like you need to make sure `LibXML.so` links with `-lirc`?

**Comment by vanzod** [comment]
@boegel I believe the problem is that the linking is done with `gcc` instead of `icc`:
```
LD_RUN_PATH="/opt/easybuild/software/Compiler/GCCcore/6.4.0/libxml2/2.9.4/lib:/opt/easybuild/software/Compiler/GCCcore/6.4.0/zlib/1.2.11/lib:/opt/easybuild/software/Compiler/GCCcore/6.4.0/XZ/5.2.3/lib" gcc  -shared -O2 -L/usr/local/lib -fstack-protector-strong Av_CharPtrPtr.o Devel.o LibXML.o dom.o perl-libxml-mm.o perl-libxml-sax.o xpath.o  -o blib/arch/auto/XML/LibXML/LibXML.so  \
   -L/opt/easybuild/software/Compiler/GCCcore/6.4.0/libxml2/2.9.4/lib -lxml2 -L/opt/easybuild/software/Compiler/GCCcore/6.4.0/zlib/1.2.11/lib -lz -L/opt/easybuild/software/Compiler/GCCcore/6.4.0/XZ/5.2.3/lib -llzma -lm -lm -lpthread -ldl
```
The problem is now how to fix this. In the `Makefile.PL` I cannot identify what generates that command line, but my Perl knowledge is pretty basic.

**Comment by boegel** [comment]
@vanzod Spot on. I was using `Perl-5.26.0-intel-2017b.eb` to resolve the `Perl` dependency for `XML-LibXML-2.0132-intel-2017b-Perl-5.26.0.eb`, i.e. I was not using `--minimal-toolchains` (which leads to using `Perl-5.26.0-GCCcore-6.4.0.eb` instead.

I was able to reproduce the issue you reported when using `--minimal-toolchains`, and was able to fix it as suggested earlier, i.e. using:

```diff
--- a/easybuild/easyconfigs/x/XML-LibXML/XML-LibXML-2.0132-intel-2017b-Perl-5.26.0.eb
+++ b/easybuild/easyconfigs/x/XML-LibXML/XML-LibXML-2.0132-intel-2017b-Perl-5.26.0.eb
@@ -22,6 +22,8 @@ dependencies = [
     ('libxml2', '2.9.4'),
 ]

+buildopts = 'LD="$CC"'
+
 options = {'modulename': 'XML::LibXML'}

 sanity_check_paths = {

**Comment by vanzod** [comment]
@boegel Do you think this is something limited to this specific package or could it be something more generic that may be worth to address on an easyblock?

**Comment by boegel** [comment]
@vanzod That's a good question, I have no idea...

**Comment by boegel** [comment]
fixed in #7440
